aquadoggo/src/schema/static_provider.rs

```diff
@@ -0,0 +1,192 @@
+    /// soon as this gets dropped.
+    ///
+    /// With the help of unsafe code this circumvents borrow-checking as we "own an unowned value".
+    #[allow(unsafe_code)]
+    pub fn new(value: T) -> Self {
+        // Leak the inner value, from now on the value is not owned anymore. If we loose the
+        // reference, the value will still exist in memory, this could potentially cause a memory
+        // leak!
+        let inner = Box::leak(Box::new(value));
+
+        // Without the knowledge of the borrow-checker we "reclaim" the ownership and keep it
+        // inside the struct. Whenever the struct gets dropped, we can safely drop the Box and also
+        // free the used memory.
+        let _ptr = unsafe { Box::from_raw(inner) };
+
+        Self { inner, _ptr }
+    }
+
+    /// Returns a reference to the value T with 'static lifetime.
+    ///
+    /// Warning: The borrow-checker "assumes" that this value will exist until the end of the
+    /// program, but that might not be the case (see examples above)! Do only use it if you either:
+    ///
+    /// 1. Make sure this value will actually be used until the end of the program (it is truly
+    ///    'static).
+    ///
+    /// 2. Make sure you do not refer to it anymore _after_ you've dropped the `StaticLeak`
+    ///    instance.
+    pub fn unsafe_value(&self) -> &'static T {
+        self.inner
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use rstest::rstest;
+
+    use p2panda_rs::schema::{FieldType, Schema, SchemaId};
+    use p2panda_rs::test_utils::fixtures::schema;
+
+    use super::{load_static_schemas, save_static_schemas, StaticLeak};
+
+    #[rstest]
+    fn save_and_load_schemas(#[from(schema)] schema_id: SchemaId) {
+        // Create and store schemas in global storage
+        let schema =
+            Schema::new(&schema_id, "Test schema", vec![("name", FieldType::String)]).unwrap();
+        let schemas = vec![schema];
+        save_static_schemas(&schemas);
+
+        // Load from global storage
+        let from_store = load_static_schemas();
+        assert_eq!(&schemas, from_store);
+    }
+
+    #[test]
+    fn drops_static_value() {
+        let managed_leak = StaticLeak::new(100);
+
+        // So far, everything is fine. We have a value with a static lifetime, which is exactly
+        // what we wanted to have.
+        let value: &'static usize = managed_leak.unsafe_value();
+        assert_eq!(*value, 100);
+
+        // We drop the struct before the program ended, this will free the used memory for the
+        // struct and the inner value! We do not have a memory leak!
+        drop(managed_leak);
+
+        // Now it gets bad .. `value` still exists in scope and the borrow checker doesn't warn us
+        // that it got dropped because it still assumes it has 'static lifetime. But actually it is
+        // gone and you will probably not find the value `100` here anymore ..
+        assert_ne!(*value, 100);
```

Comment:
This test is pretty cool :D 

But it fails sometimes for me

```
running 1 test
thread 'schema::static_provider::tests::drops_static_value' panicked at 'assertion failed: `(left != right)`
  left: `100`,
 right: `100`', aquadoggo/src/schema/static_provider.rs:194:9
```

What do?

Reply:
Could the problem be that `100` is not placed in allocated memory because it's just a teensy integer that implements `Copy`? If I replace the value with `"Hello".to_string()` the test fails with

```
error: test failed, to rerun pass '-p aquadoggo --lib'

Caused by:
  process didn't exit successfully: `/Users/work/code/zoo/aquadoggo/target/debug/deps/aquadoggo-dced19c1730dfd87 'schema::static_provider::tests::drops_static_value' --exact --nocapture` (signal: 11, SIGSEGV: invalid memory reference)
```

Which seems appropriate!

Reply:
Haha, lol, maybe your computer manages memory different? Would need to read up on that, but I doubt it has something to do with the Copy trait (I guess it's even lower since it works on Linux) 🤔